### DISCUSSION TITLE
How can I get all the qr codes that need to be validated for automated validation?

### DISCUSSION THREAD

**Discussion by JChrist** [skounis/eu-dcc-validation#113] [General]
Hello.
We need to perform the Validation Cycle 6, but currently the webapp does not provide any way to get the qr codes to be validated, so that our automated script can fill in all required information.

Currently, the best way I found to automate this was to:
* select that everything is valid, 
* download the results as json
* find through the network tab which QR codes the webapp displays (currently it is: https://github.com/eu-digital-green-certificates/dcc-quality-assurance/tree/ValidationCycle6) 
* Perform the automated scanning
* Update the json file with the actual results

**Comment by skounis** [comment]
Hi @JChrist 

The application pulls the QR codes from the following link:

* https://raw.githubusercontent.com/eu-digital-green-certificates/dcc-quality-assurance/validation6/qa.json

Does this work for your automated process. Is any other piece you need that is missing?

**Reply by JChrist** [reply]
Hi @skounis.
Yes, this is indeed what I need!
What would be helpful (for next validation cycles) is if the webapp offered some kind of why to find this file (e.g. if it showed a link to it, or at least include it in the email along with the token).
Also, it would be good if the webapp offered a way to get the expected results with empty information, ready to be filled-in.

**Reply by skounis** [reply]
Captured
* https://github.com/skounis/eu-dcc-validation/issues/117

**Comment by skounis** [comment]
@JChrist Please bear in mind to combine your results with the details provided in the header part of this request:

https://api.github.com/repos/eu-digital-green-certificates/dcc-quality-assurance/git/trees/validation6?recursive=1

![image](https://user-images.githubusercontent.com/690575/151778322-469df427-328f-4116-8627-8cfd5f1e3cbb.png)

**Reply by JChrist** [reply]
Yeah, I thought as much, so I got the "sample" results json file by selecting that everything was valid and then clicking the download button.
